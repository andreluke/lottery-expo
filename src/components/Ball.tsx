import React from 'react';
import { View, Text } from 'react-native';
import { styles } from '../theme';

interface BallProps {
  number: number;
  color: string;
  text: string,
  spacing: number;
}

const Ball: React.FC<BallProps> = ({ number, color, text, spacing }) => {
  return (
    <View style={[styles.ball, { backgroundColor: color, margin: spacing }]}>
      <Text style={[styles.number, {color: text}]}>{number}</Text>   
    </View>
  );
};

export default Ball;
