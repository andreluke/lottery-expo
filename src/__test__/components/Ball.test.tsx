import React from 'react';
import renderer from 'react-test-renderer';
import { View, Text } from 'react-native'; // Importando View e Text
import Ball from '../../components/Ball'; // Atualize o caminho se necessário

describe('Ball Component', () => {
  it('Deve renderizar corretamente com suas props', () => {
    const tree = renderer.create(
      <Ball number={7} color="blue" text="white" spacing={10} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Deve aplicar corretamente os seus estilos de background e espaço', () => {
    const component = renderer.create(
      <Ball number={7} color="blue" text="white" spacing={15} />
    );
    const ballInstance = component.root.findByType(View); // Usando View de 'react-native'

    expect(ballInstance.props.style).toContainEqual({ backgroundColor: 'blue', margin: 15 });
  });

  it('Deve aplicar corretamente a cor do texto', () => {
    const component = renderer.create(
      <Ball number={7} color="blue" text="white" spacing={10} />
    );
    const textInstance = component.root.findByType(Text); // Usando Text de 'react-native'

    expect(textInstance.props.style).toContainEqual({ color: 'white' });
  });

  it('Deve apresentar corretamente o numero', () => {
    const component = renderer.create(
      <Ball number={7} color="blue" text="white" spacing={10} />
    );
    const textInstance = component.root.findByType(Text);

    expect(textInstance.props.children).toBe(7);
  });
});
