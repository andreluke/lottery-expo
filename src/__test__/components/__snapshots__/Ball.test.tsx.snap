// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ball Component Deve renderizar corretamente com suas props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "margin": 4,
        "width": 50,
      },
      {
        "backgroundColor": "blue",
        "margin": 10,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "bold",
        },
        {
          "color": "white",
        },
      ]
    }
  >
    7
  </Text>
</View>
`;
